helix-term/src/ui/editor.rs

```diff
@@ -638,22 +638,23 @@ impl EditorView {
         // avoid lots of small allocations by reusing a text buffer for each line
         let mut text = String::with_capacity(8);
 
-        for (constructor, width) in view.gutters() {
-            let gutter = constructor(editor, doc, view, theme, is_focused, *width);
-            text.reserve(*width); // ensure there's enough space for the gutter
+        for gutter_type in view.gutters() {
+            let gutter = gutter_type.row_styler(editor, doc, view, theme, is_focused);
+            let width = gutter_type.width(view);
```

Comment:
`width` is no longer passed to `gutter_type.row_styler` (formerly `constructor`), though I could see the argument for calculating `width` first and providing it to the `row_styler`.